vignettes/speedup_options.Rmd

```diff
@@ -0,0 +1,458 @@
+
+```{r runtimes}
+# Extract the run times
+runtimes <- lapply(
+  results,
+  function(x) ifelse(is.null(x$error), round(as.duration(x$result$timing), 1), NA)
+)
+
+# Convert to a table format
+
+runtimes_dt <- melt(
+  as.data.table(runtimes),
+  measure.vars = names(model_configs),
+  variable.name = "model",
+  value.name = "runtime"
+)
+
+# Add model descriptions
+
+runtimes_dt <- merge(
+  runtimes_dt,
+  model_descriptions,
+  by = "model"
+)
+
+setcolorder(runtimes_dt, c("model", "description", "runtime"))
+
+# Order by run time
+
+runtimes_dt <- runtimes_dt[order(runtime), ]
+
+# Print table
+knitr::kable(runtimes_dt, caption = "Run times for various _EpiNow2_ models")
+```
+
+### Estimation performance
+
+Now, we will compare the estimated and true values using the continuous ranked probability score (CRPS). The CRPS is a proper scoring rule that measures the accuracy of probabilistic forecasts. We will use the `crps()` function from the `{scoringutils}` package.
+
+To calculate the CRPS for the estimated R_t and infections, we will first set up a function that makes sure the true data and estimates are of the same length and calls the `crps_sample()` function from the `{scoringutils}` package.
+```{r crps-func}
+# A function to calculate the CRPS
+calc_crps <- function(x, truth) {
+  shortest_obs_length <- min(ncol(x), length(truth))
+  reduced_truth <- tail(truth, shortest_obs_length)
+  reduced_x <- tail(t(x), shortest_obs_length)
+  return(crps_sample(reduced_truth, reduced_x))
+}
+```
+
+Now, we will extract the $R_t$ and infection estimates and calculate the CRPS using the `calc_crps()` function above.
+
+```{r fit_crps}
+# Function to extract Rt estimates
+Rt_estimated <- lapply(results, function(x) {
+  if (is.null(x$error)) {
+    obj <- x$result$estimates$fit
+    if (inherits(obj, "stanfit")) {
+      if ("R[1]" %in% names(obj)) {
+        extract(obj, "R")$R
+      } else {
+        extract(obj, "gen_R")$gen_R
+      }
+    } else {
+      obj |>
+        as_draws_matrix() |>
+        subset_draws(variable = "R")
+    }
+  } else {
+    NA
+  }
+})
+# CRPS for the Rt estimates
+rt_crps <- lapply(
+  Rt_estimated,
+  calc_crps,
+  truth = R
+)
+
+# Function to extract infection estimates
+infections_estimated <- lapply(results, function(x) {
+  if (is.null(x$error)) {
+    obj <- x$result$estimates$fit
+    if (inherits(obj, "stanfit")) {
+      extract(obj, "infections")$infections
+    } else {
+      obj |>
+        as_draws_matrix() |>
+        subset_draws(variable = "infections")
+    }
+  } else {
+    NA
+  }
+})
+
+# CRPS for the infections estimates
+infections_crps <- lapply(
+  infections_estimated,
+  calc_crps,
+  truth = infections_true
+)
+```
+
+We will now post-process the CRPS results to make them easier to visualise by adding a  "time" column and reshaping the data to long format.
+
+```{r fit_postprocessing}
```

Comment:
I'd like to see scores summarised by period (with evidence, partial evidence, and forecast) in order more easily evidence the recommendations below.

Reply:
These have now been included. Please see the latest version in 495881519f2368faed9080eda7532640e4d20d14 and let me know what you think.